pyro/distributions/improper_uniform.py

```diff
@@ -0,0 +1,67 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+from torch.distributions import constraints
+
+from .torch_distribution import TorchDistribution
+from .util import broadcast_shape
+
+
+class ImproperUniform(TorchDistribution):
```

Comment:
Just to make sure that you already consider it: should it be better to use the shorter name `Improper`?

Reply:
Hmm... I guess there are other improper distributions, e.g. `Unit`.